internal/services/cdn/cdn_frontdoor_route_resource_test.go

```diff
@@ -0,0 +1,319 @@
+)
+
+type CdnFrontDoorRouteResource struct{}
+
+// TODO: Add test with real custom domain once I figure out how to do that
+// in the meantime I will use the link_to_default_domain_enabled to make the
+// Frontdoor endpoint our "Custom Domain".
+
+func TestAccCdnFrontDoorRoute_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cdn_frontdoor_route", "test")
+	r := CdnFrontDoorRouteResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("cdn_frontdoor_origin_group_id", "cdn_frontdoor_origin_ids"),
+		{
+			// You must delete the route first to disassociate the endpoint from the origin and origin group
+			Config: r.destroy(data),
+			Check:  acceptance.ComposeTestCheckFunc(),
+		},
+	})
+}
+
+func TestAccCdnFrontDoorRoute_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cdn_frontdoor_route", "test")
+	r := CdnFrontDoorRouteResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+		{
+			// You must delete the route first to disassociate the endpoint from the origin and origin group
+			Config: r.destroy(data),
+			Check:  acceptance.ComposeTestCheckFunc(),
+		},
+	})
+}
+
+func TestAccCdnFrontDoorRoute_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cdn_frontdoor_route", "test")
+	r := CdnFrontDoorRouteResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("cdn_frontdoor_origin_group_id", "cdn_frontdoor_origin_ids"),
+		{
+			// You must delete the route first to disassociate the endpoint from the origin and origin group
+			Config: r.destroy(data),
+			Check:  acceptance.ComposeTestCheckFunc(),
+		},
+	})
+}
+
+func TestAccCdnFrontDoorRoute_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cdn_frontdoor_route", "test")
+	r := CdnFrontDoorRouteResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("cdn_frontdoor_origin_group_id", "cdn_frontdoor_origin_ids"),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("cdn_frontdoor_origin_group_id", "cdn_frontdoor_origin_ids"),
+		{
+			// You must delete the route first to disassociate the endpoint from the origin and origin group
+			Config: r.destroy(data),
+			Check:  acceptance.ComposeTestCheckFunc(),
+		},
```

Comment:
(as above) this is a design issue, we should look to introduce an `association` resource for this which'd mean we can create/remove these in a single apply

Reply:
Fixed.